### ISSUE TITLE
Backup setup

### ISSUE THREAD

**Issue by ixjlyons** [moorepants/resonance#30]
Deployment should be basically fully automated, so mainly student notebooks and the nbgrader database and such should be backed up.

Look into:

- [chrashplan](http://itcatalog.ucdavis.edu/service/crashplan)
- dream objects
- our own NAS

- [x] Jason create account on NAS for backup
- [x] https://andidittrich.de/2016/03/howto-re-enable-scpssh-login-on-synology-dsm-6-0-for-non-admin-users.html
- [x] Kenny setup key pair and give public to Jason
- [x] Setup cronjob to backup home on the server using rsync
- [x] Kenny look into what should be backup
- [x] Add cron + rsync tasks to Ansible deployment
- [x] Write a script to zip up home folder with date
- [x] Give instructions to students to back up their files using the script

**Comment by ixjlyons** [comment]
Made a little backup script to find all of their files and add them to a zip file:

```bash
#!/bin/bash
homedir=$(getent passwd "$LOGNAME" | cut -d: -f6)
backupfile="backup.zip"
find "$homedir" ! -name "$backupfile" | zip -u "$homedir/$backupfile" -@
```

and added instructions to the website: https://moorepants.github.io/eng122/pages/software.html#backing-up-your-work

**Comment by moorepants** [comment]
@ixjlyons What's the status here? Do we have a complete backup system running?

**Comment by ixjlyons** [comment]
The backup setup we currently have just uses rsync to backup every user's data. The Ansible deployment ideally makes it easy to get things back up and running within a day if something goes wrong (assuming you have a machine to use). I haven't *actually tested* recovering from backup, though.

There's [this issue](https://github.com/ixjlyons/jupyterhub-deploy-teaching/issues/5) I filed early on to integrate the backup setup into the deployment, but I never really wrapped my head around how to do it cleanly, securely, etc. I wrote some notes on how to do it manually in the README [here](https://github.com/ixjlyons/jupyterhub-deploy-teaching/blob/8e13799e61d8dffd3673336556c52c63ee973ca9/README.rst#backup).
